target-app/Home/Views/BottomSheetContainerViewController.swift

```diff
@@ -0,0 +1,196 @@
+        
+        switch sender.state {
+        case .began, .changed:
+            if self.state == .full {
+                guard translation.y > 0 else { return }
+                
+                topConstraint.constant = -(configuration.height - yTranslationMagnitude)
+                
+                self.view.layoutIfNeeded()
+            } else {
+                let newConstant = -(configuration.initialOffset + yTranslationMagnitude)
+                
+                guard translation.y < 0 else { return }
+                guard newConstant.magnitude < configuration.height else {
+                    self.showBottomSheet()
+                    return
+                }
+                
+                topConstraint.constant = newConstant
+                
+                self.view.layoutIfNeeded()
+            }
+        case .ended:
+            if self.state == .full {
+                
+                if velocity.y < 0 {
+                    // Bottom Sheet was full initially and the user tried to move it to the top
+                    self.showBottomSheet()
+                } else if yTranslationMagnitude >= configuration.height / 2 || velocity.y > 1000 {
+                    self.hideBottomSheet()
+                } else {
+                    self.showBottomSheet()
+                }
+            } else {
+                
+                if yTranslationMagnitude >= configuration.height / 2 || velocity.y < -1000 {
+                    
+                    self.showBottomSheet()
+                    
+                } else {
+                    self.hideBottomSheet()
+                }
+            }
+        case .failed:
+            if self.state == .full {
+                self.showBottomSheet()
+            } else {
+                self.hideBottomSheet()
+            }
+        default: break
+        }
+    }
+    
+    // MARK: - Configuration
+    public struct BottomSheetConfiguration {
+        let height: CGFloat
+        let initialOffset: CGFloat
+    }
+    
+    private let configuration: BottomSheetConfiguration
+    
+    // MARK: - State
+    public enum BottomSheetState {
+        case initial
+        case full
+    }
+    
+    var state: BottomSheetState = .initial
+    
+    // MARK: - Children
+    let contentViewController: Content
+    let bottomSheetViewController: BottomSheet
+    
+    // MARK: - Top Constraint
+    private var topConstraint = NSLayoutConstraint()
+    
+    // MARK: - Pan Gesture
+    lazy var panGesture: UIPanGestureRecognizer = {
+        let pan = UIPanGestureRecognizer()
+        pan.delegate = self
+//        pan.addTarget(self, action: #selector(handlePan))
+        return pan
+    }()
+    
+    // MARK: - UI Setup
+    private func setupUI() {
+        self.addChild(contentViewController)
+        self.addChild(bottomSheetViewController)
+        
+        self.view.addSubview(contentViewController.view)
+        self.view.addSubview(bottomSheetViewController.view)
```

Comment:
Avoid explicit use of self when possible.

Reply:
done